modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2661,7 +2661,7 @@ def is_reduce_fn(fn, deep_level=0):
                 by = [by]
         else:
             if not isinstance(by, list):
-                by = [by]
+                by = [by] if by is not None else []
```

Comment:
Should this be handled at the API layer?

Reply:
I don't think so. This line is the small part of correct `by` calculation in this block of code in `PandasQueryCompiler.groupby_agg` method. Also, this is general calculation for all API methods which use `groupby_agg` implementation.

Reply:
I am still thinking this might be moved to the API layer, but if we do this, we will get an error from Omnisci QC `NotImplementedError("unsupported groupby args")`. To support the case for Omnisci we should probably try to do this in a separate PR.

Reply:
If we really want to move `by` parameter handling to the API layer it should be done in a separate PR.